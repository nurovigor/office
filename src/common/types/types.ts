export type OpenSpaceNumber = 1 | 2 | 3 | 4;

export type LinkType = {
	path: string;
	title: string;
};

export type RotateType = 0 | 90 | 180 | 270;

export type DirectionType = 'top' | 'right' | 'bottom' | 'left';

export type Nullable<T> = null | T;

export type FilterName = {
	name: string;
	type: string;
	bind: string;
};

export type UpdateDataType = {
	developerId: Nullable<string>;
	tableId: Nullable<string>;
	pc: Nullable<string>;
	monitor: Nullable<string>;
	keyboard: Nullable<string>;
	mouse: Nullable<string>;
	microphone: Nullable<string>;
	headphones: Nullable<string>;
	camera: Nullable<string>;
};

export type GetTechnics = {
	name: string;
	type: string;
	bind: string;
	currentPage: number;
	pageSize: number;
	sort: string[];
};

export interface TechnicTypeI {
	bind: string;
	name: Nullable<string>;
	serial: Nullable<string>;
	type: Nullable<string>;
	__v: number;
	_id: string;
}

export type FilterType = {
	name: string;
	selectedOption: string;
	options: string[];
};

export type CreateDataResponse = {
	name: string;
	type: string;
	serial: string;
	bind: string;
};
