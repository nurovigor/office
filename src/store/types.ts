import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { store } from './store';

//types
export type AppStateType = ReturnType<typeof store.getState>;
//AppState type
export type AppDispatch = ThunkDispatch<AppStateType, unknown, AppActionType>;
//thunks type
export type AppThunk<ReturnType = void> = ThunkAction<
	ReturnType,
	AppStateType,
	unknown,
	AppActionType
>;
//actions type
export type AppActionType = any;
