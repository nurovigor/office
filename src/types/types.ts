export type OpenSpaceNumber = 1 | 2 | 3 | 4;

export type LinkType = {
	path: string;
	title: string;
};

export type RotateType = 0 | 90 | 180 | 270;

export type DirectionType = 'top' | 'right' | 'bottom' | 'left';

export type Nullable<T> = null | T;
