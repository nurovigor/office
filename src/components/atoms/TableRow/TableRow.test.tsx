import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { TableRow } from './';
import { Nullable, TechnicTypeI } from 'src/common/types/types';

describe('TableRow component', () => {
	let data: TechnicTypeI;
	let selectedId: Nullable<string>;
	let changeSelectedItem: (itemId: Nullable<string>) => void;
	beforeEach(() => {
		data = {
			bind: false,
			name: 'Panasonic',
			serial: null,
			type: 'Headphones',
			__v: 1,
			_id: '1'
		};
		selectedId = '1';
		changeSelectedItem = jest.fn();
	});
	it('renders TableRow component', () => {
		const { asFragment } = render(
			<table>
				<TableRow data={data} selectedId={selectedId} changeSelectedItem={changeSelectedItem} />
			</table>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('render with binding', () => {
		data = {
			bind: true,
			name: 'Panasonic',
			serial: null,
			type: 'Headphones',
			__v: 1,
			_id: '1'
		};
		render(
			<table>
				<TableRow data={data} selectedId={selectedId} changeSelectedItem={changeSelectedItem} />
			</table>
		);
		expect(screen.getByTestId('bind')).toHaveTextContent(/Yes/i);
	});

	it('onClick works', () => {
		render(
			<table>
				<TableRow data={data} selectedId={selectedId} changeSelectedItem={changeSelectedItem} />
			</table>
		);
		fireEvent(
			screen.getByTestId('table-row'),
			new MouseEvent('click', {
				bubbles: true,
				cancelable: true
			})
		);
		expect(changeSelectedItem).toBeCalledWith('1');
	});
});
