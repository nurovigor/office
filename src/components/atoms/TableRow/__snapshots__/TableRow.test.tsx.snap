// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow component renders TableRow component 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        data-testid="table-row"
      >
        <td>
          Panasonic
        </td>
        <td>
          Headphones
        </td>
        <td
          data-testid="bind"
        >
          No
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
